Fix add site modal cross position for RTL languages

sejas opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

- Fixes STU-820

## Proposed Changes

- Hide the window controls on Mac to fix the close cross button for RTL and LTR languages in add site modal
- Fix the position of the close button for Windows on LTR languages. Hiding the window controls would require a big refactor.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Run `npm start`
- Change the Studio language to RTL form the settings menu
- Click on `Add site` in the sidebar
- Observe the close button is on the left

| Before | After |
|--------|--------|
|  <img width="2138" height="1424" alt="Screenshot 2025-09-26 at 01 43 23" src="https://github.com/user-attachments/assets/bcf58017-313a-443c-ae25-a63ad6be41a7" /> | <img width="2138" height="1424" alt="Screenshot 2025-09-26 at 13 05 04" src="https://github.com/user-attachments/assets/c784609e-6ea7-44b1-879b-d0724ee4e2e0" /> |
| <img width="1800" height="1274" alt="Screenshot 2025-09-26 at 13 23 59" src="https://github.com/user-attachments/assets/5e8c6b24-d999-4b30-8d8a-f048d4ea1805" /> | <img width="1796" height="1270" alt="Screenshot 2025-09-26 at 13 04 07" src="https://github.com/user-attachments/assets/99604514-61f4-4ce7-b545-0cef58649861" /> |




## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Have you checked for TypeScript, React or other console errors?

sejas committed:
Fix add site modal cross position for rtl languages

bcotrim reviewed (commented):
Thanks for making the changes @sejas 
Would it possible to hide the window controls when that modal is open?

gcsecsey reviewed (approved):
Thanks @sejas, this LGTM and works well. 👍 

<img width="1984" height="1424" alt="Image" src="https://github.com/user-attachments/assets/a82a02f8-0971-44f8-b8f1-69b5d6eaf93c" />

sejas committed:
Hide electron controls when full screen modal is open

sejas committed:
Rename window to parentWindow for consistency

sejas committed:
Remove the extra top margin for close button for rtl because now the window controls are hidden

sejas committed:
Avoid drag so the button is clickable

sejas committed:
Add space top to fullscreen modal close button for ltr windows

sejas committed:
Fixed tests

sejas commented:
@bcotrim , @gcsecsey , thanks for reviewing it.

I added a logic to hide the window controls on Mac when the full screen is open https://github.com/Automattic/studio/pull/1820/commits/0cd4ceb94781a6ebdf105fe352d44ee7c1246316.
Implementing a similar logic for Windows seems not possible unless we implement the window controls in the React side, which probably is not worth it.

**MAC**

<img width="2138" height="1424" alt="Screenshot 2025-09-26 at 13 05 04" src="https://github.com/user-attachments/assets/cf7db1d2-326f-4791-a7f0-caa3d586a317" />

I observed that for Windows the close button was also conflicting with the window controls in LTR, so I fixed that too by increasing the margin top.

**Windows**

| Before | After |
|--------|--------|
|  <img width="1800" height="1274" alt="Screenshot 2025-09-26 at 13 23 59" src="https://github.com/user-attachments/assets/d530a10a-6379-414f-8e5d-78ab26b83972" /> |  <img width="1796" height="1270" alt="Screenshot 2025-09-26 at 13 04 07" src="https://github.com/user-attachments/assets/a69c5f9f-dab2-4730-a746-a576be3e666e" /> |

bcotrim commented:
thanks for adding that change @sejas 
That still looks a bit strange on Windows, but I think we can address that in a follow-up
